src/Container.ts

```diff
@@ -89,8 +92,16 @@ export class Container {
             params.unshift(null);
         }
 
-        // create a new instance of the requested object
-        const objectInstance = new (type.bind.apply(type, params))();
+        let objectInstance: any;
+
+        if (factory) {
+            // Calling factory function if specified to create an instance.
+            objectInstance = factory.apply(undefined, params);
```

Comment:
but Im not sure about this design. Problem with this design is if someday we will need to pass anything else (for example container instance, or some options) we can't do it